homeassistant/components/ovo_energy/sensor.py

```diff
@@ -32,199 +52,148 @@ async def async_setup_entry(
 
     if coordinator.data:
         if coordinator.data.electricity:
-            entities.append(OVOEnergyLastElectricityReading(coordinator, client))
-            entities.append(
-                OVOEnergyLastElectricityCost(
+            entities = [
+                *entities,
+                OVOEnergySensor(
+                    coordinator,
+                    OVOEnergySensorEntityDescription(
+                        key=f"{DOMAIN}_{client.account_id}_last_electricity_reading",
+                        name="OVO Last Electricity Reading",
+                        device_class=DEVICE_CLASS_ENERGY,
+                        state_class=STATE_CLASS_TOTAL_INCREASING,
+                        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+                        value=lambda usage: usage.electricity[-1].consumption,
+                    ),
```

Comment:
It would be better to move the descriptions to a constant tuple, instead of having them inline.
I.e. `SENSOR_TYPES_ELECTRICITY` and `SENSOR_TYPES_GAS`. Both with `tuple[OVOEnergySensorEntityDescription, ...]`.

(See comments below for specific issues with the move.)

Reply:
As a followup: With this change, you can use a list comprehension to construct the entities:
```py
[
    OVOEnergySensor(name_prefix, coordinator, description)
    for description in SENSOR_TYPES_ELECTRICITY
]
```